### ISSUE TITLE
Monterey M1: Permission denied (os error 13)

### ISSUE THREAD

**Issue by EdmundsEcho** [tokio-rs/prost#660]
Hello, 

This `.proto` compiler  is "great stuff" - great features, e.g., porting the Rust comments and mapping compounded package names (foo.bar -> foo::bar).  Details that make a world of difference.

In my "fumbling-up" the learning curve, I was letting `prost` use/build `protoc`.  It was working.  In trying to problem solve elsewhere, I went ahead and followed a recommendation to built and install `protoc` (to speed the build process).  

### The issue

I can invoke the newly installed `protoc` from the terminal (e.g., compile the proto files into python with out_dir=src/grpc ).  However, when I run `cargo build --bin myproject-server`, I'm getting the following error:

```rs
Error: Custom { kind: PermissionDenied, error: "failed to invoke protoc (hint: https://docs.rs/prost-build/#sourcing-protoc): Permission denied (os error 13)" }
```

The `target/debug/build/proc-macro2..`,  `target/debug/build/prost-build...` and `target/debug/myproject...` directories are being generated.  The `../out` directories for each are empty.

### Context

I'm running Monterey on M1 silicon.  The permissions, as far as I can tell seem ok.  I also deleted what I thought were all of the proto-related files in the build directory (in the event the new protoc was being denied the ability to overwrite the old).

Here are the permissions for `protoc`:

```fish
-rwxr-xr-x  1 root  wheel  38083 May 29 10:30 protoc
```

I've also made sure that `protoc` has access to my files and folders.

Is there anything else that I can check?

\- E

## Update

The access issue seems to be avoided when I point `tonic::build` to a `protoc` instance installed by way of homebrew.  I'm having difficulty not attributing this to something to do with `tonic::build` because I could compile the `.proto` files using the subject of this post.  Any ideas?

**Comment by EdmundsEcho** [comment]
What might I systematically look for in the permissions.

In other words, "as long as the following permissions are in place, build.rs + tonic_build should work".

**Comment by LucioFranco** [comment]
I am not sure but you could reproduce it via writing a rust binary that invokes protoc to see if that works.

**Comment by LucioFranco** [comment]
Going to close this for now since prost works on my mac m1.

**Comment by fercele** [comment]
In my machine (Windows 11), the permission denied error was due to my "PROC" environment variable pointing to a folder, instead of to the "protoc"binary itself. What i did was change the value for the env var, instead of putting the folder where I unzipped protoc, I put the fully qualified path to the binary. Worked right away. The message in windows is misleading. Permission denied. Of course it is denied, Rust was trying to execute a folder. lol. Hope this helps. Spent a bunch of hours trying to figure this out.
